core/commands/add.go

```diff
@@ -30,9 +30,11 @@ const (
 
 var AddCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
-		Tagline: "Add a file or directory to ipfs.",
+		Tagline: "Add a file to ipfs.",
 		ShortDescription: `
-Adds contents of <path> to ipfs. Use -r to add directories (recursively).
+Adds contents of <path> to ipfs. Use -r to add directories.
+Note that directories are added recursively, to form the ipfs
+MerkleDAG.
```

Comment:
- I think this is also a regression. I dont think this language change adds, it only adds more words, while the meaning is still the same. 
- Also "to form the ipfs MerkleDAG", "the" is off. "to form an ipfs MerkleDAG" sounds better.
- And, files are also formed as merkledags, so is everything you add to ipfs. I think this clause is more confusing than helpful. I do agree that explaining _what_'s going on is a good idea. This is something for the Long Description.

I would vote for either:

```
Adds contents of <path> to ipfs. Use -r to add directories, recursively.
```

or 

```
Adds contents of <path> to ipfs. Use -r to add directories, recursively.

Files and directories will be chunked and imported into ipfs. File chunks
and directory entries will be linked with cryptographic hashes, forming
an ipfs merkledag. For more information about how file importing works
and the merkledag data structures, see <link>.
```